src/parser.rs

```diff
@@ -353,37 +357,46 @@ impl<'input, Endian> DebugInfo<'input, Endian>
 /// An iterator over the compilation- and partial-units of a section.
 ///
 /// See the [documentation on
-/// `DebugInfo::units`](./struct.DebugInfo.html#method.units)
-/// for more detail.
+/// `DebugInfo::units`](./struct.DebugInfo.html#method.units) for more detail.
 pub struct UnitHeadersIter<'input, Endian>
     where Endian: Endianity
 {
     input: EndianBuf<'input, Endian>,
 }
 
-impl<'input, Endian> Iterator for UnitHeadersIter<'input, Endian>
+impl<'input, Endian> UnitHeadersIter<'input, Endian>
     where Endian: Endianity
 {
-    type Item = ParseResult<UnitHeader<'input, Endian>>;
-
-    fn next(&mut self) -> Option<Self::Item> {
+    /// Advance the iterator to the next unit header.
+    pub fn next(&mut self) -> ParseResult<Option<UnitHeader<'input, Endian>>> {
         if self.input.is_empty() {
-            None
+            Ok(None)
         } else {
             match parse_unit_header(self.input) {
```

Comment:
I don't feel strongly about it, but we could change this to `try!(parse_unit_header(self.input)`. That would have the effect that we don't update `self.input` on error though.